agent/engine/execcmd/manager.go

```diff
@@ -115,3 +116,12 @@ func IsExecEnabledContainer(container *apicontainer.Container) bool {
 	_, ok := container.GetManagedAgentByName(ExecuteCommandAgentName)
 	return ok
 }
+
+// HasExecAgentInitFailed returns if exec feature has been disabled for a container
```

Comment:
don't think this needs to use negated logic, personally I think it's clearer to use something like this:

```go
func ExecAgentInitSucceeded(ma) bool {
    // check
}
```

and the caller of `ExecAgentInitSucceeded` can get `ma.Reason` if this checker returns `false`

Reply:
I kept the name HasExecAgentInitFailed because I am checking exactly that in this method. That is why, I thought this will be more apt. But let me know if you feel that I should change the name

Reply:
sounds good! other than the name, it would still make it easier to understand by returning only the result (`bool`) instead of a tuple of `(string, bool)`; consider the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)...

Reply:
Changed it.